### PR TITLE
feat: create EthrDIDs and set DID owner (#1)

### LINKED ISSUE #1: HU: Se puede crear nuevos DID ETHR y agregarlo al ETHR Registry
**Tareas:**

- [x] Desplegar ETHR Registry en Amoy
- [x] Implementar libreria @kaytrust/did-ethr
- [x] Desarrollar los Unit Test
- [x] Crear repositorio en Github
- [x] Crear pipeline en Github Actions
- [x] Crear documentación en README.md

**Criterios de aceptación:**

1. La libreria debe estar disponible en Github y tener la visibilidad para todo el publico
2. Se debe poder crear DID ETHR solo de forma local
3. La libreria debe ser capaz de registar el DID ETHR que cree de forma local a traves de parametros opcionales al ETHR Registry
4. La libreria debe ser capaz de registrar DID´s ETHR existentes al ETHR Registry
5. Debe estar habilitado un pipeline en Github actions que sea capaz de transpilar la aplicación a travez de un build y que ejecture las pruebas unitarias
6. La libreria debe desarrollarse en TypeScript y para las pruebas unitarias deben usarse Jest
7. La documentación debe incluir:
- Breve explicación del proposito de la libreria
- Requisitos de instalación
- Pasos para instalación 
- Ejemplos necesario para su uso, segun la cantidad de funciones
- Listado de propiedades con su descripción y valores por defecto
8. Debe desarrollarse las pruebas unitarias necesarias para cada funcionalidad de la libreria

### PR DESCRIPTION
**Pull request by GeinerGV**
- Implemented functionality to create EthrDIDs and test cases
- Added capability to set the owner of a DID and test cases

### CODE REVIEW

File: .github/workflows/npm-publish.yml
Comment (nullbeam) [inline .github/workflows/npm-publish.yml:1]: Can you run this pipeline to test which functions are ok?

File: index.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+export * from './src'
+
+type EnvVars = Partial<{
+    RPC_AMOY: string,
+    CONTRACT_REGISTRY_AMOY: string,
+    RPC_CARDONA: string,
+    CONTRACT_REGISTRY_CARDONA: string,
+    TEST_KEY_1: string
+    TEST_KEY_2: string
+    // TEST_WITH_CHANGE_OWNER: string|number|boolean
```
Comment (nullbeam) [inline index.ts]: What is this?

File: src/constants.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export const RPC_AMOY_DEFAULT = process.env.RPC_AMOY || "https://polygon-amoy.drpc.org"
+export const CONTRACT_REGISTRY_AMOY_DEFAULT = process.env.CONTRACT_REGISTRY_AMOY || "0xBC56d0883ef228b2B16420E9002Ece0A46c893F8"
```
Comment (nullbeam) [inline src/constants.ts:3]: I think we should not have default addresses or urls, but rather show an error message in case it is not configured.

Comment (GeinerGV) [reply src/constants.ts:3]: They are only for unit tests, that default URL is used for a test to create a did and use that network to communicate with the contract. The same with the address registry that I will use to do the tests on Amoy network.

File: index.ts (hunk 1 above)
Comment (GeinerGV) [reply index.ts]: It is used for typescript to recognize the environment variables available in process.env, these variables allow us to customize the rpcs that will be used for testing, replace hard-coded public rpcs with "private" rpcs and make tests faster many times and thus avoid the limitations of public rpcs.

Also add private keys to test transactions in the contract, such as changing the owner of the did ethr, etc.

Comment (nullbeam) [reply index.ts]: I understand, but my doubt was more about why these variables are mentioned and what they do.

File: src/constants.ts (hunk 2 above)
Comment (nullbeam) [reply src/constants.ts:3]: Okay

File: index.ts (hunk 1 above)
Comment (GeinerGV) [reply index.ts]: I pushed a file (`types.d.ts`) that has comments explaining some things about ENV VARS

**Review by nullbeam — APPROVED** [review]

**Pull request merged**